testing/file_test/test_file.cpp

```diff
@@ -673,61 +702,94 @@ auto ProcessTestFile(llvm::StringRef test_name, bool running_autoupdate)
-                                     &test_file.expected_stderr));
-    if (is_consumed) {
-      continue;
-    }
-
-    // At this point, lines are retained as non-CHECK lines.
-    test_file.non_check_lines.push_back(
-        FileTestLine(split.file_index, line_index, line));
+    if (test_file) {
+      CARBON_ASSIGN_OR_RETURN(
+          is_consumed, TryConsumeCheck(running_autoupdate, line_index, line,
+                                       line_trimmed, test_file->expected_stdout,
+                                       test_file->expected_stderr));
+      if (is_consumed) {
+        continue;
+      }
 
-    CARBON_ASSIGN_OR_RETURN(
-        is_consumed, TryConsumeArgs(line, line_trimmed, &test_file.test_args,
-                                    &test_file.extra_args));
-    if (is_consumed) {
-      continue;
+      // At this point, lines are retained as non-CHECK lines.
+      test_file->non_check_lines.push_back(
+          FileTestLine(split_state.file_index, line_index, line));
     }
-    CARBON_ASSIGN_OR_RETURN(
-        is_consumed,
-        TryConsumeIncludeFile(line_trimmed, &test_file.include_file_splits));
-    if (is_consumed) {
-      continue;
-    }
-    CARBON_ASSIGN_OR_RETURN(
-        is_consumed,
-        TryConsumeAutoupdate(line_index, line_trimmed, &found_autoupdate,
-                             &test_file.autoupdate_line_number));
+
+    CARBON_ASSIGN_OR_RETURN(is_consumed,
+                            TryConsumeArgs(line, line_trimmed, args));
     if (is_consumed) {
       continue;
     }
-    CARBON_ASSIGN_OR_RETURN(
-        is_consumed,
-        TryConsumeSetFlag(line_trimmed, "SET-CAPTURE-CONSOLE-OUTPUT",
-                          &test_file.capture_console_output));
+    CARBON_ASSIGN_OR_RETURN(is_consumed,
+                            TryConsumeExtraArgs(line_trimmed, extra_args));
     if (is_consumed) {
       continue;
     }
     CARBON_ASSIGN_OR_RETURN(is_consumed,
-                            TryConsumeSetFlag(line_trimmed, "SET-CHECK-SUBSET",
-                                              &test_file.check_subset));
+                            TryConsumeIncludeFile(line_trimmed, include_files));
     if (is_consumed) {
       continue;
     }
+
+    if (test_file) {
+      CARBON_ASSIGN_OR_RETURN(
+          is_consumed,
+          TryConsumeAutoupdate(line_index, line_trimmed, *found_autoupdate,
+                               test_file->autoupdate_line_number));
+      if (is_consumed) {
+        continue;
+      }
+      CARBON_ASSIGN_OR_RETURN(
+          is_consumed,
+          TryConsumeSetFlag(line_trimmed, "SET-CAPTURE-CONSOLE-OUTPUT",
+                            test_file->capture_console_output));
+      if (is_consumed) {
+        continue;
+      }
+      CARBON_ASSIGN_OR_RETURN(
+          is_consumed, TryConsumeSetFlag(line_trimmed, "SET-CHECK-SUBSET",
+                                         test_file->check_subset));
+      if (is_consumed) {
+        continue;
+      }
```

Comment:
Do you think it's worth detecting these markers (and `// CHECK`) in included files so we can produce errors on them?

Reply:
Thanks!

One more, if you like: would it make sense to check that an included file doesn't define a split named `AUTOUPDATE-SPLIT`? (I doubt this really matters so I'm fine if you want to go ahead with this as-is.)